fiona/__init__.py

```diff
@@ -67,21 +67,62 @@
 import os
 import sys
 import warnings
-
+import platform
 from six import string_types
 from collections import OrderedDict
-
-try:
-    from pathlib import Path
-except ImportError:  # pragma: no cover
-    class Path:
-        pass
+from pathlib import Path
 
 # TODO: remove this? Or at least move it, flake8 complains.
 if sys.platform == "win32":
     libdir = os.path.join(os.path.dirname(__file__), ".libs")
     os.environ["PATH"] = os.environ["PATH"] + ";" + libdir
 
+try:
+
+    from fiona.ogrext import _bounds, _listlayers, FIELD_TYPES_MAP, _remove, \
+        _remove_layer
+
+except ImportError as e:
+    """
+    With Python >= 3.8 on Windows directories in PATH are not automatically
+    searched for DLL dependencies and must be added manually with
+    os.add_dll_directory.
+
+    see https://github.com/Toblerity/Fiona/issues/851
+
+    We check if a */gdal/bin directory is found in PATH and
+    if none is found if GDAL_HOME is set.
+    """
+    if platform.system() == 'Windows' and (3, 8) <= sys.version_info:
+
+        dll_directory = None
+
+        # Parse PATH for gdal/bin
+        for path in os.getenv('PATH', '').split(';'):
+            p = Path(path.lower())
+
+            if p.parts[-2:] == ('gdal', 'bin') and os.path.exists(path):
+                dll_directory = path
+                break
+            del path, p
+
+        # Use GDAL_HOME if present
+        if dll_directory is not None:
+            gdal_home = os.getenv('GDAL_HOME', None)
```

Comment:
This will need to be added to the documentation. I don't think it's a standard runtime environment variable for GDAL, although it does seem to be used in the build guide for Windows.